app/controllers/contact_details_controller.rb

```diff
@@ -11,8 +11,10 @@ def submit
     contact_details_saver =
       ContactDetailsSaver.new(selected_answer_store: selected_answer_store)
     contact_details_saver.save(@form)
+    result =
+      CreateRepair.new.call(answers: selected_answer_store.selected_answers)
 
-    redirect_to confirmation_path('WORK_ORDER_REFERENCE')
+    redirect_to confirmation_path(result.work_order_reference)
```

Comment:
Is it possible for the `work_order_reference` used here and `request_reference` used in `ContactDetailsWithCallbackController` to clash?

Reply:
UGH - you're totally right: the confirmation page will need to know which thing to request. This needs a rethink.